File: 002/002.py
#!/usr/bin/env python3

# By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

sum = 0
f_0, f_1 = 0, 1

while True:
 f_0, f_1 = f_1, f_0 + f_1
 if f_1 >= 4*10**6:
    break
 if f_1 % 2 == 0:
    sum += f_1

print("Solution:", sum)
  

File: 002/002.java
// By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

public class FibonacciEvenSum {
 public static void main(String[] args) {
   int f_0 = 0;
   int f_1 = 1;
   int sum = 0;
   while (f_1 < 4000000){
      int f_new = f_0 + f_1;
      if (f_new %2 ==0) {
         System.out.println(f_new);
         sum += f_new;
      }
      f_0 = f_1;
      f_1 = f_new;
   }
   return sum;
 }
}
    

